### PR TITLE
Only include table names in error_tables_metadata

### PR DESCRIPTION
**Pull request by jmcatamney**
As the error_tables_metadata file created by a failed gprestore is intended to be used with --include-table and --exclude-table filters on a subsequent restore, it should not include the names of any non-table objects that failed to be restored.

### CODE REVIEW

File: end_to_end/end_to_end_suite_test.go [hunk 1]
```diff
@@ -858,6 +858,39 @@ var _ = Describe("backup and restore end to end tests", func() {
 			Expect(tables).To(HaveLen(len(expectedErrorTablesMetadata)))
 			_ = os.Remove(files[0])
 		})
+		It(`ensure gprestore of corrupt backup with --on-error-continue only logs tables in error_tables_metadata file`, func() {
```
Comment (AJR-VMware) [inline end_to_end/end_to_end_suite_test.go:861]: Nit: we uses the grammatical convention for these.  Would be "ensures".

File: end_to_end/end_to_end_suite_test.go [hunk 2]
```diff
@@ -858,6 +858,39 @@ var _ = Describe("backup and restore end to end tests", func() {
...
+			if segmentCount != 3 {
+				Skip("Restoring from a tarred backup currently requires a 3-segment cluster to test.")
+			}
+			// The functionality works on 5, but it's currently difficult to make a saved backup on a 5 cluster.
+			// TODO: Remove this and re-do the saved backup once we get local 5 testing working again.
```
Comment (AJR-VMware) [inline end_to_end/end_to_end_suite_test.go:866]: Are we planning to get local 5 testing working again?  This seems like it'll be a permanent TODO.

File: end_to_end/end_to_end_suite_test.go (hunk 1 above)
Comment (jmcatamney) [reply end_to_end/end_to_end_suite_test.go:861]: Blame the other tests I copy-pasted.  I'll fix all three.

File: end_to_end/end_to_end_suite_test.go (hunk 2 above)
Comment (jmcatamney) [reply end_to_end/end_to_end_suite_test.go:866]: We haven't given up on making it work, as far as I know.  We can chat with the team and make a decision on that.

**Review by AJR-VMware — APPROVED** [review]

**Pull request merged**